docs/CACHING.md

```diff
@@ -0,0 +1,353 @@
+          # - matchExpressions:
+          #     - key: kubernetes.io/hostname
+          #       operator: In
+          #       values:
+          #         - ip-192-0-2-0.region-code.compute.internal
+```
+
+After deploying your workload, the Mountpoint Pod should also be deployed to the same node automatically with an local NVMe PV attached to it:
+```bash
+$ kubectl describe po -n mount-s3
+Name:                 mp-ql5rd
+Namespace:            mount-s3
+...
+Volumes:
+  ...
+  local-cache:
+    Type:          EphemeralVolume (an inline specification for a volume that gets created and deleted with the pod)
+    StorageClass:  nvme-ssd
+    Volume:
+    Labels:            s3.csi.aws.com/type=local-ephemeral-cache
+    Annotations:       <none>
+    Capacity:
+    Access Modes:
+    VolumeMode:    Filesystem
+
+$ kubectl describe pvc -n mount-s3
+Name:          mp-xt6c4-local-cache
+Namespace:     mount-s3
+StorageClass:  nvme-ssd
+Status:        Bound
+Volume:        local-pv-bfea2335
+Labels:        s3.csi.aws.com/type=local-ephemeral-cache
+Annotations:   pv.kubernetes.io/bind-completed: yes
+               pv.kubernetes.io/bound-by-controller: yes
+Finalizers:    [kubernetes.io/pvc-protection]
+Capacity:      1769Gi
+Access Modes:  RWO
+VolumeMode:    Filesystem
+Used By:       mp-xt6c4
+```
+
+Note that if there is no local NVMe available in the scheduled node, Mountpoint Pod would fail to schedule and your workload would hang in `Pending` state.
+You must ensure your workload (and therefore the Mountpoint Pod) is scheduled to a node with local NVMe available to use.
+
+Ensure to check [other configurations of Local Volume Static Provisioner](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/tree/master?tab=readme-ov-file#user-guide) including
+[Local Volume Node Cleanup Controller](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/master/docs/node-cleanup-controller.md) for volume clean up and other details.
+
+#### (Deprecated) `cache` flag via `mountOptions`
+
+With the CSI Driver v1, the Mountpoint instances were spawned on the host using `systemd`, and the `cache` flag in `mountOptions` was a relative path to the host. The cache folder also needed to exists for Mountpoint to use. We deprecated this usage, and will fallback to using [`emptyDir`](#emptyDir) with the default storage medium without any limit by default.
+
+You no longer needed to create a cache folder on the host, and the configured path will be ignored by the CSI Driver v2! We recommend customers to migrate to [`emptyDir`](#emptyDir) and specify a limit.
+
+This deprecated use of cache:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: s3-pv
+spec:
+  mountOptions:
+    - cache /cache/folder/on/host
+  csi:
+    driver: s3.csi.aws.com
+    # ...
+    volumeAttributes:
+      bucketName: amzn-s3-demo-bucket
+```
+
+will translate the following automatically by the CSI Driver v2:
```

Comment:
This kinda implies that S3 CSI Driver will actually modify customer PV automatically which is not the case. We can just say that cache folder parameter from MountOptions will be ignored and we will use `emptyDir` for cache instead.